Task2/Task2/Task2/Services/ServicesByProduct.cs

```diff
@@ -0,0 +1,146 @@
+            {
+                var obj = new Brick()
+                {
+                    ProductName = leftProduct.ProductName,
+                    PurchaseCost = Math.Round((leftProduct.PurchaseCost * leftProduct.NumberOfUnits
+                                               + rightProduct.PurchaseCost * rightProduct.NumberOfUnits)
+                                              / (leftProduct.NumberOfUnits + rightProduct.NumberOfUnits), 2),
+                    Margin = Math.Round((leftProduct.Margin * leftProduct.NumberOfUnits
+                                         + rightProduct.Margin * rightProduct.NumberOfUnits)
+                                        / (leftProduct.NumberOfUnits + rightProduct.NumberOfUnits), 2),
+                    NumberOfUnits = leftProduct.NumberOfUnits + rightProduct.NumberOfUnits
+                };
+
+                return obj;
+            }
+            else
+            {
+                throw new ProductNameException("Products have different Product Name");
+            }
+        }
+
+        public static Screw AdditionOfTwoProducts(Screw leftProduct, Screw rightProduct)
+        {
+            if (leftProduct.ProductName.Equals(rightProduct.ProductName, StringComparison.InvariantCultureIgnoreCase))
+            {
+                var obj = new Screw()
+                {
+                    ProductName = leftProduct.ProductName,
+                    PurchaseCost = Math.Round((leftProduct.PurchaseCost * leftProduct.NumberOfUnits
+                                               + rightProduct.PurchaseCost * rightProduct.NumberOfUnits)
+                                              / (leftProduct.NumberOfUnits + rightProduct.NumberOfUnits), 2),
+                    Margin = Math.Round((leftProduct.Margin * leftProduct.NumberOfUnits
+                                         + rightProduct.Margin * rightProduct.NumberOfUnits)
+                                        / (leftProduct.NumberOfUnits + rightProduct.NumberOfUnits), 2),
+                    NumberOfUnits = leftProduct.NumberOfUnits + rightProduct.NumberOfUnits
+                };
+
+                return obj;
+            }
+            else
+            {
+                throw new ProductNameException("Products have different Product Name");
+            }
+        }
+
+        public static Bolt SubtractingUnitsOfProduction(Bolt product, int numberOfUnits)
+        {
+            var obj = new Bolt()
+            {
+                ProductName = product.ProductName,
+                PurchaseCost = product.PurchaseCost,
+                Margin = product.Margin
+            };
+
+            if (product.NumberOfUnits >= numberOfUnits)
+            {
+                obj.NumberOfUnits = product.NumberOfUnits - numberOfUnits;
+            }
+            else
+            {
+                throw new ProductQuantityException("Cannot subtract more units than there are in the product.");
+            }
```

Comment:
Could we rewrite this part of code as I wrote higher?

Reply:
fixed